doc/mgmt/Docker to Host communication.md

```diff
@@ -0,0 +1,235 @@
+translating the user CLIs or REST requests to actions. Most of these actions
+perform some operation on the Redis database, but some of them require
+operations to be done on the host, i.e., outside the container. This document
+describes the host server, and translib API that are used to communicate between
+the Docker container and the host.
+
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+* The SONiC Management Framework and Telemetry containers must be able to issue
+  requests to the host, and return the responses from the host.
+* The individual applications that need access to the host must be able to
+  create a host module and easily issue requests and get responses back from the
+  host.
+* The host communication API shall be available in Translib, and shall provide
+  both synchronous and asynchronous communication methods.
+* It shall be possible to configure the identity of the Linux user accounts who have access to a D-Bus socket.
+* It shall be possible to configure containers in such a way that only certain containers (e.g. SONiC Mgmt.)
+  have access to the D-Bus socket.
+
+### 1.1.2 Configuration and Management Requirements
+
+N/A
+
+### 1.1.3 Scalability Requirements
+
+N/A
+
+### 1.1.4 Warm Boot Requirements
+
+N/A
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+
+The code will extend the existing Translib modules to provide a D-Bus based
+query API to issue requests to the host. The host service will be a Python based
+application which listens on known D-Bus endpoints.https://en.wikipedia.org/wiki/D-Bus
+
+The individual app modules can extend the host service by providing a small
+Python snippet that will register against their application endpoint.
+
+### 1.2.2 Container
+
+SONiC Management Framework, gNMI Telemetry containers
+
+### 1.2.3 SAI Overview
+
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+
+All deployments
+
+## 2.2 Functional Description
+
+This feature enables management applications to issue
+requests to the host to perform actions such as:
+* image install / upgrade
+* ZTP enable/disable
+* initiate reboot and warm reboot using existing scripts
+* create show-tech tar file using existing show-tech script
+* config save/reload using existing scripts
+
+# 3 Design
+## 3.1 Overview
+
+The feature extends the SONiC management framework to add a D-Bus service on the host. This service will register against a known endpoint, and will service requests to the endpoint. Application modules will add "applets" to the host
+service, which will automatically register their endpoints (D-Bus objects and methods). 
+
+The client application module in a container (e.g. management or telemetry) can use the APIs provided in Translib to send the request to the host - essentially, make a (remote) D-Bus method call, and either wait for the response (if the request was synchronous), or receive a channel and wait for the request to return the response on the
```

Comment:
How to secure the users shouldn't mis use the critical host access to commands like reboot? Can you provide some guidelines around it?